packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-License-Identifier: MIT
+// solhint-disable reason-string
+pragma solidity ^0.8.0;
+
+import "@manifoldxyz/creator-core-solidity/contracts/core/IERC1155CreatorCore.sol";
+import "@manifoldxyz/creator-core-solidity/contracts/extensions/ICreatorExtensionTokenURI.sol";
+
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+import "./GachaLazyClaim.sol";
+import "./IERC1155GachaLazyClaim.sol";
+
+/**
+ * @title Gacha Lazy 1155 Payable Claim
+ * @author manifold.xyz
+ * @notice
+ */
+contract ERC1155GachaLazyClaim is IERC165, IERC1155GachaLazyClaim, ICreatorExtensionTokenURI, GachaLazyClaim {
+  using Strings for uint256;
+
+  // stores mapping from contractAddress/instanceId to the claim it represents
+  // { contractAddress => { instanceId => Claim } }
+  mapping(address => mapping(uint256 => Claim)) private _claims;
+
+  // { contractAddress => { tokenId => { instanceId } }
+  mapping(address => mapping(uint256 => uint256)) private _claimTokenIds;
+
+  // Nonce usage tracking: mapping of fid to nonce to used status
+  mapping(uint256 => mapping(uint256 => bool)) private _usedNonces;
+
+  function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, AdminControl) returns (bool) {
+    return
+      interfaceId == type(IERC1155GachaLazyClaim).interfaceId ||
+      interfaceId == type(IGachaLazyClaim).interfaceId ||
+      interfaceId == type(ICreatorExtensionTokenURI).interfaceId ||
+      interfaceId == type(IAdminControl).interfaceId ||
+      interfaceId == type(IERC165).interfaceId;
+  }
+
+  constructor(address initialOwner) GachaLazyClaim(initialOwner) {}
+
+  /**
+   * See {IERC1155GachaLazyClaim-initializeClaim}.
+   */
+  function initializeClaim(
+    address creatorContractAddress,
+    uint256 instanceId,
+    ClaimParameters calldata claimParameters
+  ) external payable override adminRequired {
+    if (instanceId == 0 || instanceId > MAX_UINT_56) revert IGachaLazyClaim.InvalidInstance();
+    // Revert if claim at instanceId already exists
+    if (_claims[creatorContractAddress][instanceId].storageProtocol == StorageProtocol.INVALID)
+      revert IGachaLazyClaim.ClaimAlreadyInitialized();
+
+    // Sanity checks
+    if (claimParameters.storageProtocol != StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+    address[] memory receivers = new address[](1);
+    receivers[0] = msg.sender;
+    string[] memory uris = new string[](1);
```

Comment:
Make this a length equal to variants so that it actually reserves the correct amount for uris and amounts.  Receivers can be 1 and it should mint to all the same person.

Reply:
string[] memory uris = new string[](claimParameters.itemVariations);